### PR TITLE
feat(store-types): Add -o flag to update integration manifest with retrieved store type definition

### PR DESCRIPTION
**Pull request by m8rmclaren**
- Add `-o | --output-to-integration-manifest` flag to output retrieved store type to an Integration Manifest.
  - If manifest exists, store type is appended to store type list
  - If manifest exists and a store type with the same name is already present, the contents are overwritten with what Command returned.
  - If the manifest doesn't exist, it's created and populated _only_ with the schema and the store type definition.
- Refactored `store-type get` path to implement the `flags.Flags` and `flags.Options` interfaces.

### CODE REVIEW

**Review by spbsoluble — CHANGES_REQUESTED** [review]

File: cmd/storeTypes_get.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	storeTypeName *string
+	// genericFormat strips all fields specific to the Command instance from the output.
+	genericFormat *bool
+	// gitRef is the git branch or tag to reference when pulling store-types from the internet.
+	gitRef *string
+	// outputToIntegrationManifest updates the integration manifest with the store type. It overrides the store type in the manifest if it already exists.
+	outputToIntegrationManifest *bool
+}
+
+func NewStoreTypesGetFlags() *StoreTypesGetFlags {
```
[the remark this answers is not in the document]
Comment (m8rmclaren) [reply cmd/storeTypes_get.go]: Resolved in 3a6be4cf7c543184cc0120e18046a57349c58b4d

File: cmd/storeTypes_get.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+func (f *StoreTypesGetFlags) AddFlags(flags *pflag.FlagSet) {
+	flags.IntVarP(f.storeTypeID, "id", "i", -1, "ID of the certificate store type to get.")
+	flags.StringVarP(f.storeTypeName, "name", "n", "", "Name of the certificate store type to get.")
+	flags.BoolVarP(f.genericFormat, "generic", "g", false, "Output the store type in a generic format stripped of all fields specific to the Command instance.")
+	flags.StringVarP(f.gitRef, FlagGitRef, "b", "main", "The git branch or tag to reference when pulling store-types from the internet.")
+	flags.BoolVarP(f.outputToIntegrationManifest, "output-to-integration-manifest", "o", false, "Update the integration manifest with the store type. It overrides the store type in the manifest if it already exists. If the integration manifest does not exist in the current directory, it will be created.")
+}
+
+func NewCmdStoreTypesGet() *cobra.Command {
```
[the remark this answers is not in the document]
Comment (m8rmclaren) [reply cmd/storeTypes_get.go]: `NewCmdStoreTypesGet` is synonymous with "Instantiate a Store Types Get Cobra command" - I could change it to `CreateCmdStoreTypesGet`?

[the remark this answers is not in the document]
Comment (m8rmclaren) [reply cmd/storeTypes_get.go]: Resolved in a1cb13c330f5d4475d0867a96a6d926f4d06c4aa

**Review by m8rmclaren — COMMENTED** [review]
Thanks for the feedback

**Review by spbsoluble — CHANGES_REQUESTED** [review]
Please regenerate the docs and fix the test that is failing: https://github.com/Keyfactor/kfutil/actions/runs/6802999075/job/18497341205#step:3:93

I suspect you might need to add another couple tests as well:

1. `kfutil store-types get --name $NAME -g` no write to file
2. `kfutil store-types get --name $NAME -g -o $VALID_MANIFEST_FILE`
3. `kfutil store-types get --name $NAME -g -o $INVALID_FILE`

**Comment by m8rmclaren** [comment]
> Please regenerate the docs and fix the test that is failing: https://github.com/Keyfactor/kfutil/actions/runs/6802999075/job/18497341205#step:3:93
> 
> I suspect you might need to add another couple tests as well:
> 
> 1. `kfutil store-types get --name $NAME -g` no write to file
> 2. `kfutil store-types get --name $NAME -g -o $VALID_MANIFEST_FILE`
> 3. `kfutil store-types get --name $NAME -g -o $INVALID_FILE`

When output to manifest (`--output-to-integration-manifest`) is present, the generic format flag is automatically set. This behavior is noted in the usage.

```go
	// Set the output type to full unless genericFormat is true
	f.outputType = "full"
	if f.genericFormat {
		f.outputType = "generic"
	}

	// If outputToIntegrationManifest is true, set the output type to generic
	if f.outputToIntegrationManifest {
		f.outputType = "generic"
	}
```

`get [-i <store-type-id> | -n <store-type-name>] [-b <git-ref>] [-g | --output-to-integration-manifest]`

**Review by spbsoluble — APPROVED** [review]

**Pull request merged**